File: VB/WcfService1/Service1.svc.vb
Imports System
Imports DevExpress.Xpo

Imports DevExpress.Xpo.DB
Imports DevExpress.Xpo.DB.Helpers


Namespace WcfService1

	Public Class Service1
		Inherits CachedDataStoreService
		Public Shared MainDataStore As ICachedDataStore

		Shared Sub New()
			Dim connectionString As String = MSSqlConnectionProvider.GetConnectionString("localhost", "ServiceDB")
			Dim dataStore As IDataStore = XpoDefault.GetConnectionProvider(connectionString, AutoCreateOption.DatabaseAndSchema)
			Dim dataCacheRoot As New DataCacheRoot(dataStore)
			dataCacheRoot.Configure(New DataCacheConfiguration(DataCacheConfigurationCaching.InList, "Customer"))
			MainDataStore = dataCacheRoot
		End Sub

		Public Sub New()
			MyBase.New(MainDataStore)
		End Sub

	End Class

End Namespace

File: CS/WcfService1/Service1.svc.cs
using DevExpress.Xpo;

using DevExpress.Xpo.DB;
using DevExpress.Xpo.DB.Helpers;


namespace WcfService1 {

    public class Service1 : CachedDataStoreService {

        public static ICachedDataStore MainDataStore;

        static Service1() {
            string connectionString = MSSqlConnectionProvider.GetConnectionString("localhost", "ServiceDB");
            IDataStore dataStore = XpoDefault.GetConnectionProvider(connectionString, AutoCreateOption.DatabaseAndSchema);
            DataCacheRoot dataCacheRoot = new DataCacheRoot(dataStore);
            dataCacheRoot.Configure(new DataCacheConfiguration(DataCacheConfigurationCaching.InList, "Customer"));
            MainDataStore = dataCacheRoot;
        }

        public Service1() : base(MainDataStore) { }

    }

}
